tracer/test/Datadog.Trace.Security.Unit.Tests/WafCompatibilityTests.cs

```diff
@@ -27,5 +27,17 @@ public void ShouldNotInitializeWithExportsMissing(string version)
             libraryInitializationResult.Success.Should().BeFalse();
             libraryInitializationResult.WafLibraryInvoker.Should().BeNull();
         }
+
+        [SkippableTheory]
+        [InlineData("1.14.0")]
+        public void ShouldNotInitializeWithKnownIncompatibility(string version)
+        {
+            // for some reason, these tests cause a "DDWAF_ERROR": [ddwaf_run]interface.cpp(206): std::bad_alloc on mac when run with others
```

Comment:
IMO this needs to be investigated as it could hide a bigger problem. Also, possibly related, WafLibraryInvoker.Initialize loads the library before checking the version number, and doesn't unload it if incompatible.
Could even be worth using a SafeHandle implementation for libraryHandle to ensure we never leak it.

Reply:
The comment was copy / pasted from another test that's been there a while. I've re-enabled both tests and they don't see to be failing. I think there's a fair chance the issue has been fixed.

I have added code to close the library on failure to load, but not migrated to safe handle. See 29e2d8e